### ISSUE TITLE
[Bug]: Application menu navigation accessibility improvements

### ISSUE THREAD

**Issue by NPavie** [electron/electron#41324]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

28.2.2

### What operating system are you using?

Windows

### Operating System Version

Windows 11 23H2 build 22631.3085

### What arch are you using?

x64

### Expected Behavior

On Windows, given an application menu defined with the following snippet : 
```ts
const template: MenuItemConstructorOptions = [
 { 
    label: '&File',
    submenu: [ /*...*/ ]
  },
  { 
    label: '&View',
    submenu: [ /*...*/ ]
  }, /* ...*/
]
const menu = Menu.buildFromTemplate(template)
Menu.setApplicationMenu(menu)
```

- On opening the menu section, the first "enabled" (as clickable) element should be focused and reported as selected in the UI
- For an application menu with several sections like in the example, if the currently selected entry in a section has no sub items, left and right arrow should allow navigating to the left or right section of the menu

### Actual Behavior

Part of our users and testers are visually impaired and rely on keyboard for navigation with screen readers like NVDA.

On Windows, our testers reported 2 accessibility issues regarding our electron application menu : 
- When opening any section of the application menu with the keyboard, the "pop-up" opens under the menu button, the first element is reported by screen readers, but it is not actually selected : 
Users need to do an additionnal action (pressing enter or using arrow down or arrow right) to actually have the first enabled entry of the menu to be selected and to be clickable (without the ui reporting to screen readers that the element is now actually clickable)

Note that I noticed the same issue with chrome default behaviour regarding menus on Windows. (But edge is not concerned by this issue)

- When navigating through menus, left and right arrow keys cannot be used to switch directly from one menu section to another.
In the case of entry without submenus, left and right arrows are doing nothing, while in many Windows application, by default, the arrow keys allow navigating through the main sections.

**Comment by codebytere** [comment]
@NPavie do you happen to have an example of a native app that works as expected in this case? That'd be really helpful in ensuring correct behavior when a maintainer is able to work on this!

**Comment by NPavie** [comment]
@codebytere Windows explorer expose the described keyboard navigation.
(I don't use it much with keyboard navigation but I tested it and it seems it is this menu behavior that is expected by our visually impaired Windows users and testers)

Just in case I tested a small electron fiddle and some thing weird was happening here : 
I did a naive run, selected the app (probably by clicking on the web context) and did notice the behaviour with or without the narrator: 
- When the pop-up menu opens, it is in a kind of "no selection" state, with no item selected.
- The narrator does highlight and report the first element as being selected, but trying to activate the item with `Enter` actually enters the menu (without reporting anything vocally as if no actions were performed, only the background color of the button changes).

Then I wanted to capture my screen to make a gif and strangely enough, the menu was now working as expected.

I think i have a test scenario to better identify when the menu issue occurs: 
- if you focus on the app by clicking on (or switching with alt-tab to) the application in the taskbar, the popup menu works as expected (with the first item being selected and activable on opening)
- But If you click somewhere in the web context and then try to open the menu, then the first item is not correctly selected, and the popup menu opens in the kind of "no selection" state (while still being announced and highlighted in narrators)

Hope this helps
